x-pack/plugin/esql/src/main/antlr/EsqlBaseParser.g4

```diff
@@ -173,15 +173,20 @@ constant
     | decimalValue                                                                      #decimalLiteral
     | integerValue                                                                      #integerLiteral
     | booleanValue                                                                      #booleanLiteral
-    | PARAM                                                                             #inputParam
+    | params                                                                            #inputParams
     | string                                                                            #stringLiteral
     | OPENING_BRACKET numericValue (COMMA numericValue)* CLOSING_BRACKET                #numericArrayLiteral
     | OPENING_BRACKET booleanValue (COMMA booleanValue)* CLOSING_BRACKET                #booleanArrayLiteral
     | OPENING_BRACKET string (COMMA string)* CLOSING_BRACKET                            #stringArrayLiteral
     ;
 
+params
+    : PARAM                        #inputParam
+    | PARAM_NAMED_OR_POSITIONAL    #inputParamNamedOrPositional
+    ;
+
 limitCommand
-    : LIMIT INTEGER_LITERAL
+    : LIMIT (INTEGER_LITERAL | params)
```

Comment:
I don't have a strong opinion whether support parameters for limit or not, I left it there since it is there. But I think in case people ask why not support parameters in other places since you support it in limit, I'll remove it until there is a need to do it.